providers-sdk/v1/util/tracer/func.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package tracer
+
+import (
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+// FuncDur must be used in a `defer` statement. It receives the function name and the time
+// when a function started and logs out the time it took to run.
+//
+// ```go
+//
+//	func MyFunction() {
+//		defer tracer.FuncDur(time.Now(), "mypackage.MyFunction")
+//
+//		...
+//	}
+//
+// ```
+func FuncDur(start time.Time, name string) {
```

Comment:
@jaym what do you think about using opentelemetry instead. This would allow us to use standard tracing?

Reply:
I'm not sure how much overhead opentelemetry has, but my guess is that it would not be appropriate if you want to be measuring things in the mql execution. For things like us making remote calls, it could make sense. Would need to solve how you get spans from the providers back to the root process.

There's also https://pkg.go.dev/runtime/trace which I think should be low overhead when not tracing. And the docs say it plays nicely with go's trace tools. Not sure how we would handle forked processes.

Both these probably also require making sure context is properly handled and passed where it needs to be

Reply:
This is a minimal implementation to avoid adding much overhead, I do agree with both of you that we should use opentelemetry, can we get this in and release to continue troubleshooting of some issues we have and swap this in the future?